tracer/src/Datadog.Trace.Tools.dd_dotnet/CreatedumpCommand.cs

```diff
@@ -0,0 +1,676 @@
+            {
+                var path = IntPtr.Zero;
+
+                try
+                {
+                    AnsiConsole.WriteLine($"Writing crash report to {outputFile}...");
+                    path = Marshal.StringToHGlobalAnsi(outputFile);
+                    crashReport.WriteToFile(path);
+                }
+                finally
+                {
+                    if (path != IntPtr.Zero)
+                    {
+                        Marshal.FreeHGlobal(path);
+                    }
+                }
+            }
+            else
+            {
+                crashReport.Send();
+            }
+        }
+        catch (Win32Exception ex)
+        {
+            Errors.Add($"Failed to send crash report: {GetLastError(crashReport, ex)}");
+            return;
+        }
+
+        AnsiConsole.WriteLine("Datadog - Crash report sent successfully");
+    }
+
+    private void InvokeCreatedump(string createdumpPath)
+    {
+        if (File.Exists(createdumpPath))
+        {
+            var commandLineArgs = Environment.GetCommandLineArgs();
+
+            if (commandLineArgs.Length <= 2)
+            {
+                // It should never happen, but who knows
+                AnsiConsole.WriteLine($"Datadog - Unable to call createdump, missing arguments: {Environment.CommandLine}");
+            }
+            else
+            {
+                System.Diagnostics.Process.Start(commandLineArgs[2], commandLineArgs[3..]).WaitForExit();
+            }
+        }
+    }
+
+    private string GetLastError(ICrashReport crashReport, Win32Exception exception)
+    {
+        if (exception.NativeErrorCode == 1)
+        {
+            // libdatadog error
+            crashReport.GetLastError(out var message, out var length);
+
+            if (message == IntPtr.Zero)
+            {
+                return "unknown error";
+            }
+
+            return Marshal.PtrToStringAnsi(message, length);
+        }
+
+        return $"unspecified error: {exception.NativeErrorCode}";
+    }
+
+    private bool SetSignal(ICrashReport crashReport, int signal)
+    {
+        try
+        {
+            crashReport.SetSignalInfo(signal, IntPtr.Zero);
+        }
+        catch (Win32Exception ex)
+        {
+            Errors.Add($"Failed to set signal info: {GetLastError(crashReport, ex)}");
+            return false;
+        }
+
+        return true;
+    }
+
+    private unsafe bool SetMetadata(ICrashReport crashReport, ClrRuntime runtime, ClrException? exception)
+    {
+        var flavor = runtime.ClrInfo.Flavor switch
+        {
+            ClrFlavor.Core => ".NET Core",
+            ClrFlavor.Desktop => ".NET Framework",
+            // ClrFlavor.NativeAOT => "NativeAOT",
+            ClrFlavor f => f.ToString()
+        };
+
+        var version = string.Empty;
+
+        if (runtime.ClrInfo.Version.Major != 0)
+        {
+            version = runtime.ClrInfo.Version.ToString();
+        }
```

Comment:
Is this number "wrong" for early .NET Core versions?

Reply:
Yes it is